docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -605,6 +605,34 @@ Unbound variables are automatically pulled from the surrounding local
 and global scopes, and the result of the compilation is cached for
 efficient re-use.
 
+Compiling with ``cython.compile``
+=================================
+
+Cython supports transpartent compiling of the cython code within function using
+``@cython.compile`` dedorator::
+
+    @cython.compile
+    def plus(a, b):
+        return a + b
+
+Parameters of the decorated function cannot have types. Their types are
+automaticaly determined based on values passed to a function.
```

Comment:
```suggestion
set to be a standard Python object.
```

The way you've written it implies JIT compilation specialized for whatever type you've passed (which I don't think is what happens)

Reply:
> The way you've written it implies JIT compilation specialized for whatever type you've passed 

Well Cython does kind of specialisation if I understand it correctly - see the generated .pyx files by the decorator below. Cython generates two versions of function (`_cython_inline_4528cbcb5eb32c44847d2ed113ddee648d7485d6.pyx` and `_cython_inline_3e54f923a152621eae433e8eb5c2eb8559332453.pyx`) - one with python string and other with C long type. 


```
$ python test.py
Compiling /Users/matus/Library/Caches/cython/inline/_cython_inline_4528cbcb5eb32c44847d2ed113ddee648d7485d6.pyx because it changed.
[1/1] Cythonizing /Users/matus/Library/Caches/cython/inline/_cython_inline_4528cbcb5eb32c44847d2ed113ddee648d7485d6.pyx
warning: /Users/matus/Library/Caches/cython/inline/_cython_inline_4528cbcb5eb32c44847d2ed113ddee648d7485d6.pyx:7:4: Unreachable code
35
Compiling /Users/matus/Library/Caches/cython/inline/_cython_inline_3e54f923a152621eae433e8eb5c2eb8559332453.pyx because it changed.
[1/1] Cythonizing /Users/matus/Library/Caches/cython/inline/_cython_inline_3e54f923a152621eae433e8eb5c2eb8559332453.pyx
warning: /Users/matus/Library/Caches/cython/inline/_cython_inline_3e54f923a152621eae433e8eb5c2eb8559332453.pyx:7:4: Unreachable code
8
(cython39) matus@MacBook-Pro:~/dev/cython/test$ cat /Users/matus/Library/Caches/cython/inline/_cython_inline_4528cbcb5eb32c44847d2ed113ddee648d7485d6.pyx



def __invoke(str a, str b):

    return a + b
    return locals()
(cython39) matus@MacBook-Pro:~/dev/cytho/Users/matus/Library/Caches/cython/inline/_cython_inline_3e54f923a152621eae433e8eb5c2eb8559332453.pyx59332453.pyx



def __invoke(long a, long b):

    return a + b
    return locals()
```

Reply:
Oh OK I didn't know that. What you've written is correct then.

Reply:
No worries. It was surprising also for me :-D